src/modules/did/did.service.ts

```diff
@@ -151,19 +151,10 @@ export class DIDService {
       this.logger.info(`refreshing cached document for did: ${did}`);
       const cachedDIDDocument = await this.didRepository.findOne(did);
 
-      const logs = await this.updateLogs(
-        cachedDIDDocument.id,
-        this.parseLogs(cachedDIDDocument.logs),
-      );
-
-      const updatedDidDocument = this.resolveDocumentFromLogs(
-        cachedDIDDocument.id,
-        logs,
-      );
-
+      const logs = await this.getAllLogs(did);
```

Comment:
Full sync by the time potentially might take enormous amount of time and number of requests to chain. Perhaps it should be one-time operation possibly triggered from some endpoint

Reply:
I see two ways:
1. Special script for that
2. Endpoint by only authorized for user with some specific role claim

Reply:
Also may be it is possible to add `readBatch` in did-reg

Reply:
I agree with Dmitry that we shouldn't change the approach permanently to be a full sync instead of incremental. IMO, a good approach would be if we can control via configuration whether or not a full vs incremental sync of DID docs is performed.